src/StagedProposalProcessor.sol

```diff
@@ -450,14 +483,18 @@ contract StagedProposalProcessor is IProposal, PluginUUPSUpgradeable {
             // the remaining 1/64 gas are sufficient to successfully finish the call.
             uint256 gasBefore = gasleft();
 
+            // TODO: in the createProposal standardization, shall we rename it to `data` instead of `metadata` ? 
+            // This way, people would understand that it could be anything.
```

Comment:
`data` might be misunderstood as the function call data, maybe description?

Reply:
In the `IProposal`, since it's gonna get generic, some people might want to pass some arbitrary encoded things in that parameter in `createProposal` - forget about SPP but in general case. They want to inherit from `IProposal` and pass encoded `data` and then they will decode it inside `createProposal`. This allows such contracts to be able to receive any kind of extra data they wish so. If we name it `description`, they start thinking: ah, it's `description`, so it can not receive some custom data we want ?

I am not sold on this. Would love your counter arguments.

Reply:
ohh, got your point now